src/proto/FunctionRpc.proto

```diff
@@ -117,13 +117,35 @@ message WorkerInitRequest {
 
 // Worker responds with the result of initializing itself
 message WorkerInitResponse {
-  // Version of worker
+  // NOT USED
+  // TODO: Remove from protobuf during next breaking change release
   string worker_version = 1;
+
   // A map of worker supported features/capabilities
   map<string, string> capabilities = 2;
 
   // Status of the response
   StatusResult result = 3;
+
+  // Worker metadata captured for telemetry purposes
+  WorkerMetadata worker_metadata = 4;
+}
+
+message WorkerMetadata {
+  // The runtime/stack name
```

Comment:
some information is captured in `worker.config.json` - example - https://github.com/Azure/azure-functions-python-worker/blob/dev/python/prodV4/worker.config.json - just FYI

Reply:
We will use the worker.config file as a fallback for the `language` and potentially the `runtime_version`, making some of thee properties optional. E.g. for most workers we can probably use `language` from the worker config, but for .net we cannot differentiate between .NET and .NET framework so the field here comes in handy